### DISCUSSION TITLE
Auto compression in zip files

### DISCUSSION THREAD

**Discussion by thoiruddin** [tgdrive/teldrive#31] [Q&A]
Hi,

I read about maximum files size, it is 2GB right?
So, if I have 1 folder which contained files around 8GB, can teldrive auto split and zip it into 2GB?

Also, can I upload multiple folders (which 1 folders have files more than 2 GB) and store it in single upload?

**Comment by drumpinicho** [comment]
Teldrive auto-splits everything you upload. You don't need to do anything, although you can specify the size of the splits.

If the file you upload is 8GB, the result will be 4 files of 2GB or 8 files of 1GB... in your telegram, depending on how you configure the split size.

If you want to upload everything compressed, you can use the uploader or rclone through a pipe

**Reply by thoiruddin** [reply]
Hi,

Thanks for your reply. The uploader your mention is this one isn't?

https://github.com/divyam234/teldrive-upload

And can you let me know about providers I need to use if I use rclone? There is a lot of providers on rclone site list.
Which one will you choose?

Sorry for edit this.

For teldrive-upload, it's work. But, I try to upload all in one folder. How to make it?

To make it clear, here is the structure.

-Folder A
--Subfolder
--Files

and upload all in one zip files with name of Folder A.

**Reply by drumpinicho** [reply]
There is no magic formula to compress a file or folder before uploading it
You can compress it manually and then upload it via teldrive or the uploader or using the command line with:

`zip -5 -r file.zip path/to/compress && /path/to/uploader -path "/path/to/file.zip" -dest "/wherever/you/want"`

To use rclone, you have to use the unofficial client created by divyam234 and which is in his github, in which you can select teldrive as provider
